Looks like your non-linux codepath doesn't compile quite yet:

```
gcc -DHAVE_CONFIG_H -I. -I..     -g -O2 -Wall -I/Users/andi/dev/oflops/../openflow/include -Wshadow -MT myargs.o -MD -MP -MF .deps/myargs.Tpo -c -o myargs.o myargs.c
mv -f .deps/myargs.Tpo .deps/myargs.Po
gcc -DHAVE_CONFIG_H -I. -I..     -g -O2 -Wall -I/Users/andi/dev/oflops/../openflow/include -Wshadow -MT cbench.o -MD -MP -MF .deps/cbench.Tpo -c -o cbench.o cbench.c
cbench.c:446:111: error: too few arguments to function call, expected 9, have 8
        fakeswitch_init(&fakeswitches[i], sock, 65536, debug, delay, mode, total_mac_addresses, learn_dst_macs);
        ~~~~~~~~~~~~~~~                                                                                    
```

--- a/cbench/cbench.c
+++ b/cbench/cbench.c
@@ -23,7 +23,7 @@
 #include "cbench.h"
 #include "fakeswitch.h"
 
-#ifdef __linux__
+#ifdef USE_EPOLL
 #include <sys/epoll.h>
 #define MAX_EVENTS	16
 struct epoll_event events[MAX_EVENTS];
@@ -74,7 +74,7 @@ double run_test(int n_fakeswitches, struct fakeswitch * fakeswitches, int mstest
         if( (1000* diff.tv_sec  + (float)diff.tv_usec/1000)> total_wait)
             break;
 
-        #ifdef __linux__
+        #ifdef USE_EPOLL
         for(i = 0; i < MAX_EVENTS; i++) {
             events[i].events = EPOLLIN | EPOLLOUT;
         }
@@ -158,7 +158,7 @@ int timeout_connect(int fd, const char * hostname, int port, int mstimeout) {
 		return -1;
 	}
 	
-    #ifdef __linux__
+    #ifdef USE_EPOLL
     struct epoll_event ev;
     int epollfd = epoll_create(1);
     ev.events = EPOLLIN | EPOLLOUT | EPOLLERR;
@@ -188,7 +188,7 @@ int timeout_connect(int fd, const char * hostname, int port, int mstimeout) {
 			    return -1;
 		    }
         }
-        #ifdef __linux__
+        #ifdef USE_EPOLL
 	    int nfds = epoll_wait(epollfd, &ev, 1, mstimeout);
         #else
         ret = select(fd + 1, NULL, &fds, NULL, &tv);
@@ -196,7 +196,7 @@ int timeout_connect(int fd, const char * hostname, int port, int mstimeout) {
 	}
     freeaddrinfo(res);
 
-    #ifdef __linux__
+    #ifdef USE_EPOLL
     close(epollfd);
 	
     if(ev.events & EPOLLERR) {
@@ -413,7 +413,7 @@ int main(int argc, char * argv[])
     double  v;
     results = malloc(tests_per_loop * sizeof(double));
 
-    #ifdef __linux__
+    #ifdef USE_EPOLL
     struct epoll_event ev;
     epollfd = epoll_create(4096);
     if(epollfd == -1) {
@@ -440,17 +440,17 @@ int main(int argc, char * argv[])
         if(debug)
             fprintf(stderr,"Initializing switch %d ... ", i+1);
         fflush(stderr);
-        #ifdef __linux__
+        #ifdef USE_EPOLL
         fakeswitch_init(&fakeswitches[i],dpid_offset+i,sock,BUFLEN, debug, delay, mode, total_mac_addresses, learn_dst_macs);
         #else
-        fakeswitch_init(&fakeswitches[i], sock, 65536, debug, delay, mode, total_mac_addresses, learn_dst_macs);
+        fakeswitch_init(&fakeswitches[i], 0, sock, 65536, debug, delay, mode, total_mac_addresses, learn_dst_macs);
         #endif
 
         if(debug)
             fprintf(stderr," :: done.\n");
         fflush(stderr);
     
-        #ifdef __linux__
+        #ifdef USE_EPOLL
 	    ev.events = EPOLLIN | EPOLLOUT;
 	    ev.data.fd = sock;
 	    ev.data.ptr = &fakeswitches[i];

--- a/cbench/cbench.h
+++ b/cbench/cbench.h
@@ -5,4 +5,8 @@
 #define BUFLEN 65536
 #endif
 
+#ifdef __linux__
+#define USE_EPOLL
+#endif
+
 #endif

--- a/cbench/fakeswitch.c
+++ b/cbench/fakeswitch.c
@@ -19,7 +19,7 @@
 #include "cbench.h"
 #include "fakeswitch.h"
 
-#ifdef __linux__
+#ifdef USE_EPOLL
 #include <sys/epoll.h>
 #endif
 
@@ -61,7 +61,7 @@ void fakeswitch_init(struct fakeswitch *fs, int dpid, int sock, int bufsize, int
     struct ofp_header ofph;
     fs->sock = sock;
     fs->debug = debug;
-    #ifdef __linux__
+    #ifdef USE_EPOLL
     fs->id = dpid;
     #else
     static int ID = 1;
@@ -527,7 +527,7 @@ static void fakeswitch_handle_write(struct fakeswitch *fs)
 /***********************************************************************/
 void fakeswitch_handle_io(struct fakeswitch *fs, void *pfd_events)
 {
-    #ifdef __linux__
+    #ifdef USE_EPOLL
     int events = *((int*) pfd_events);
     if(events & EPOLLIN) {
         fakeswitch_handle_read(fs);

